src/main/java/functional/algebraic/Result.java

```diff
@@ -454,12 +501,35 @@ public void match(Consumer<? super E> ef, Consumer<? super V> vf)
      */
     public <T> Result<E, T> map(Function<? super V, ? extends T> f)
     {
-        return match(
+        return matchThen(
                 (E e) -> error(e),
                 (V v) -> value(f.apply(v))
         );
     }
 
+    /**
+     * Maps a function of errors to the error inside this Result, or does nothing if there is none.
+     * @param f the function to apply
+     * @param <F> the resulting error type
+     * @return the mapped Result
+     */
+    public <F extends Exception> Result<F, V> mapError(Function<? super E, ? extends F> f) {
+        return matchThen(
+                (E e) -> error(f.apply(e)),
+                (V v) -> value(v)
+        );
+    }
+
+    /**
+     * Applies a function to turn an error into a value if there is an error, and returns the resulting value
+     * or the value inside this Result.
+     * @param f the function
+     * @return
+     */
+    public V consumeError(Function<? super E, ? extends V> f) {
```

Comment:
Given the name similarity to `Maybe::consume`, and the fact that this does not involve a `Consumer`, maybe `convert`, `transform` or `transmute` would be better?
```suggestion
    public V convertError(Function<? super E, ? extends V> f) {
```